experiment_prototype/scan_classes/sequences.py

```diff
@@ -337,107 +260,97 @@ def __init__(self, seqn_keys, sequence_slice_dict, sequence_interface, transmit_
-                if pulse_index == 0:
-                    # calculate the first rx sample and set the value.
-                    self.first_rx_sample_time = calculate_first_rx_sample_time(
-                        pulse_samples[0].shape[0], txrate)
-                # Can plot for testing here
-                # plot_samples('channel0.png', pulse_samples[0])
-                # plot_fft('fftplot.png', pulse_samples[0], prog.txrate)
-
-            # This is all the data required for a pulse.
-            pulse_transmit_data['startofburst'] = startofburst
-            pulse_transmit_data['endofburst'] = endofburst
-            pulse_transmit_data['pulse_antennas'] = pulse_antennas
-            pulse_transmit_data['samples_array'] = pulse_samples
-            pulse_transmit_data['timing'] = timing
-            pulse_transmit_data['isarepeat'] = repeat
-
-            # Add pulse dictionary pulse_transmit_data at last place in sequence list
-            sequence_list.append(pulse_transmit_data)
-
-        self.find_blanks()
-
-        return sequence_list
+                samples = np.repeat(basic_samples[np.newaxis,:,:], num_pulses, axis=0)
+
+            # sum the samples into their position in the sequence buffer.
+            tr_window_num_samps = round(tr_window_time * txrate)
+            for i, pulse in enumerate(self.combined_pulses_metadata):
+                for component_info in pulse['component_info']:
+                    if component_info['slice_id'] == slice_id:
+                        pulse_timing_us = component_info['start_time_us']
+                        pulse_sample_start = round(txrate * (pulse_timing_us * 1e-6))
+                        pulse_samples_len = samples.shape[-1]
+
+                        start = tr_window_num_samps + pulse_sample_start
+                        end = start + pulse_samples_len
+                        pulse_piece = sequence[...,start:end]
+
+                        np.add(pulse_piece, samples[i], out=pulse_piece)
+
+        # copy the encoded and combined samples into the metadata for the sequence.
+        pulse_data = []
+        for i, pulse in enumerate(self.combined_pulses_metadata):
+            pulse_timing_us = pulse['start_time_us']
+            pulse_sample_start = round(txrate * (pulse_timing_us * 1e-6))
 
+            pulse_len = pulse['total_pulse_len']
+            num_samples = round(txrate * (pulse_len * 1e-6))
+            start = pulse_sample_start
+            end = start + num_samples + 2 * tr_window_num_samps
+            samples = sequence[...,start:end]
+
+            new_pulse_info = copy.deepcopy(pulse['transmit_metadata'])
+            new_pulse_info['samples_array'] = samples
+
+            if i != 0:
+                last_pulse = pulse_data[i-1]['samples_array']
+                if samples.shape == last_pulse.shape:
+                    if np.isclose(samples, last_pulse).all():
+                            new_pulse_info['isarepeat'] = True
```

Comment:
think the indentation is incorrect here

Reply:
Whats the concern here? It looks correct to me.

Reply:
```suggestion
                        new_pulse_info['isarepeat'] = True
```